src/main/java/picocli/CommandLine.java

```diff
@@ -5557,24 +5560,34 @@ public int size() {
 
             private final LinkedHashMap<K, V> targetMap = new LinkedHashMap<K, V>();
             private final HashMap<K, K> keyMap = new HashMap<K, K>();
-            private final Locale locale;
-            private final Set<K> keySet;
+            private final Set<K> keySet = new CaseAwareKeySet();
             private boolean caseInsensitive = false;
+            private final Locale locale;
 
             /**
-             * Constructs an empty LinkedCaseAwareMap instance with {@link java.util.Locale#ENGLISH}.
+             * Constructs an empty {@code CaseAwareLinkedMap} instance with {@link java.util.Locale#ENGLISH}.
              */
             public CaseAwareLinkedMap() {
                 this(ENGLISH);
             }
 
             /**
-             * Constructs an empty LinkedCaseAwareMap instance with the specified {@link java.util.Locale}.
+             * Constructs an empty {@code CaseAwareLinkedMap} instance with the specified {@link java.util.Locale}.
              * @param locale the locale to convert character cases
              */
             public CaseAwareLinkedMap(Locale locale) {
                 this.locale = locale;
-                this.keySet = new CaseAwareKeySet();
+            }
+
+            /**
+             * Constructs a {@code CaseAwareLinkedMap} instance with the same mappings as the specified map.
+             * The {@code CaseAwareLinkedMap} instance is created with a default locale {@link java.util.Locale#ENGLISH}.
+             * @param map the map whose mappings are to be placed in this map
+             * @throws NullPointerException if the specified map is null
+             */
+            public CaseAwareLinkedMap(Map<? extends K, ? extends V> map) {
+                this(ENGLISH);
```

Comment:
I propose we change this constructor to accept a `CaseAwareLinkedMap` instead of a plain `Map`.
In the implementation, we can then copy the `locale`, `caseInsensitive`, and other attributes from the passed-in  `CaseAwareLinkedMap` parameter. That makes the main use case easiest to use.